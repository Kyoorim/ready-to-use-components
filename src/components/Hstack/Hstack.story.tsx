import React, { FC } from 'react';
import { Meta } from '@storybook/react';
import Hstack from './Hstack';
import { CustomProps } from 'components/Stack';
import Box from '../Box/Box';

export default {
  title: 'Hstack',
  component: Hstack,
} as Meta<CustomProps>;

export const Default: FC = () => {
  return (
    <Hstack>
      <Box backgroundColor={'red'}>1</Box>
      <Box backgroundColor={'green'}>2</Box>
      <Box backgroundColor={'blue'}>3</Box>
    </Hstack>
  );
};
