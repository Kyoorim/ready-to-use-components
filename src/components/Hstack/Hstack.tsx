import React from 'react';
import Stack from '../Stack/Stack';
import { CustomProps } from '../Stack/Stack';

const HStack: FCC<CustomProps> = ({ direction, gap, ...props }) => {
  return <Stack data-testid="hstack-comp" direction="row" gap={gap} {...props}></Stack>;
};

export default HStack;
