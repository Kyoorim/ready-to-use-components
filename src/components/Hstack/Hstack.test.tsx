import React from 'react';
import { render, screen } from '@testing-library/react';
import Hstack from './Hstack';

const defaultProps = {
  onClick: jest.fn(),
  children: 'Hstack',
};

describe('<Hstack />', () => {
  it('should render', () => {
    render(<Hstack {...defaultProps} />);

    expect(screen.getByTestId('hstack-comp')).toMatchSnapshot();
  });

  it('should render children', () => {
    render(<Hstack {...defaultProps} />);

    expect(screen.getByText(/Hstack/i)).toBeInTheDocument();
  });

  it('should allow custom className', () => {
    const props = {
      ...defaultProps,
      className: 'Custom',
    };
    render(<Hstack {...props} />);

    expect(screen.getByTestId('hstack-comp')).toHaveProperty('className');
  });

  it('should only allow row stacks', () => {
    render(<Hstack {...defaultProps} />);

    expect(screen.getByTestId('hstack-comp')).toHaveStyle('flex-direction: row');
  });
});
