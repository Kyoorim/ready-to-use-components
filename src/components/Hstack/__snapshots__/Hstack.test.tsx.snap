// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hstack /> should render 1`] = `
<div
  class="sc-bcXHqe hA-dBqr"
  data-testid="hstack-comp"
  direction="row"
>
  Hstack
</div>
`;
